pkg/iterator/iterator.go

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package iterator
+
+// VecIterator is an iterator that returns a vector of elements of type T for
+// each iteration.
+type VecIterator[T any] interface {
```

Comment:
VecIterator seems like an internal implementation detail for CartesianProduct. Is exposing this type really necessary?